### PR TITLE
Clarify our version compatibility guarantees

### PR DESCRIPTION
**Pull request by zmb3**
Our upgrade procedure has always been such that servers (proxy/auth)
should be upgraded before upgrading nodes or other agents. This is
because servers can tolerate old clients, but servers do not necessarily
support newer clients (this sometimes works, but is not a supported
configuration).

While this has been the case in practice, it's not entirely clear in
our docs. The docs just say "applies to clients and servers" and never
gave a specific example of a new client trying to connect to an old
server.

Also update the versioning RFD to be more accurate with reality
(we no longer maintain release branches for minor versions).

### CODE REVIEW

File: docs/pages/setup/operations/upgrading.mdx [hunk 1]
```diff
@@ -38,11 +38,16 @@ compatible.
...
-- The above applies to both clients and servers. For example, a 6.x.x Proxy Service is
-  compatible with 5.x.x Nodes and 5.x.x `tsh`. But we don't guarantee that a
-  7.x.x `tsh` will work with a 5.x.x Proxy Service.
+- **Patch and minor** versions are always compatible, for example, any 8.0.1
+  component will work with any 8.0.3 component and any 8.1.0 component will work
+  with any 8.3.0 component.
+- Major versions are always compatible with the **previous** major release. This
+  means you must not attempt to upgrade from 6.x.x straight to 8.x.x. You must
+  upgrade to 7.x.x first.
+  - Servers support clients that are 1 major version behind, but do not support
```
Comment (ptgott) [inline docs/pages/setup/operations/upgrading.mdx:47]: What would you think of replacing the second bullet with this bullet? The third bullet is not much longer than the second bullet, but is more helpful and specific while providing the same general information.

File: docs/pages/setup/operations/upgrading.mdx [hunk 2]
```diff
@@ -38,11 +38,16 @@ compatible.
...
+    clients that are on a newer major version. For example, a 8.x.x Proxy service
```
Comment (ptgott) [inline docs/pages/setup/operations/upgrading.mdx:48]:
```suggestion
    clients that are on a newer major version. For example, a 8.x.x Proxy Service
```

File: docs/pages/setup/operations/upgrading.mdx [hunk 3]
```diff
@@ -38,11 +38,16 @@ compatible.
...
+    is compatible with 7.x.x Nodes and 7.x.x `tsh`, but we don't guarantee that a
+    9.x.x Node will with with an 8.x.x Proxy Service.
```
Comment (ptgott) [inline docs/pages/setup/operations/upgrading.mdx:50]:
```suggestion
    9.x.x Node will work with an 8.x.x Proxy Service.
```

File: docs/pages/setup/operations/upgrading.mdx (hunk 1 above)
Comment (zmb3) [reply docs/pages/setup/operations/upgrading.mdx:47]: Sure!

Comment (zmb3) [reply docs/pages/setup/operations/upgrading.mdx]: I combined them but kept the note about not upgrading from 6 to 8, as that's important.

File: docs/pages/setup/operations/upgrading.mdx [hunk 4]
```diff
@@ -64,16 +71,18 @@ When upgrading a single Teleport cluster:
 >
   If several Auth Service instances are running in the High Availability configuration
   (for example, in an AWS Auto Scaling group), you must shrink the group to
-  **just one Auth Service** before performing an upgrade. 
-  
-  While Teleport will attempt to perform any necessary migrations, we recommend users create a backup of their backend before upgrading the Auth Server as a
-  precaution. This allows for a safe rollback in case the migration itself fails.
+  **just one Auth Service** before performing an upgrade.
+
+While Teleport will attempt to perform any necessary migrations, we recommend users create a backup of their backend before upgrading the Auth Server as a
```
Comment (hatched) [inline docs/pages/setup/operations/upgrading.mdx:76]:
```suggestion
While Teleport will attempt to perform any necessary migrations, we strongly recommend users create a backup of their backend before upgrading the Auth Server as a
```

**Review by hatched — APPROVED** [review]

File: docs/pages/setup/operations/upgrading.mdx (hunk 4 above)
Comment (zmb3) [reply docs/pages/setup/operations/upgrading.mdx:76]: Meh, I think I'll leave it as is. It doesn't read much differently, and I don't expect anyone to change their behavior based on whether we recommend or strongly recommend something.

**Review by smallinsky — APPROVED** [review]

File: docs/pages/setup/operations/upgrading.mdx [hunk 5]
```diff
@@ -38,11 +38,15 @@ compatible.
...
-- **Patch and minor** versions are always compatible, for example, any 5.0.1 component will work with any 5.0.3 component and 6.1.0 component will work with any 6.7.0 component.
-- Major versions are always compatible with the **previous** major release. This means you must not attempt to upgrade from 5.x.x straight to 7.x.x. You must upgrade to 6.x.x first.
-- The above applies to both clients and servers. For example, a 6.x.x Proxy Service is
-  compatible with 5.x.x Nodes and 5.x.x `tsh`. But we don't guarantee that a
-  7.x.x `tsh` will work with a 5.x.x Proxy Service.
+- **Patch and minor** versions are always compatible, for example, any 8.0.1
+  component will work with any 8.0.3 component and any 8.1.0 component will work
+  with any 8.3.0 component.
+- Servers support clients that are 1 major version behind, but do not support
+  clients that are on a newer major version. For example, a 8.x.x Proxy Service
```
Comment (ptgott) [inline docs/pages/setup/operations/upgrading.mdx:45]:
```suggestion
  clients that are on a newer major version. For example, an 8.x.x Proxy Service
```

**Review by ptgott — APPROVED** [review]

**Pull request merged**